### PR TITLE
[wrangler] fix(secret): don't create Worker for delete-only `secret bulk` payload

### LINKED ISSUE #14052: [wrangler] Delete-only `secret bulk` input can create a new Worker
## What happens

`wrangler secret bulk` now accepts `null` values so a bulk payload can delete secrets.

If the target Worker name does not exist, a delete-only payload can still fall into the existing draft-Worker creation retry path.

That means a command like this can create a new Worker instead of failing:

```sh
echo '{"OLD_SECRET": null}' | wrangler secret bulk --name typo-worker
```

## Expected

If the bulk payload only deletes secrets and the Worker does not exist, Wrangler should leave the Worker-not-found error alone.

## Actual

Wrangler treats Worker-not-found as a reason to create a draft Worker, even though there are no secrets to create or update.

In non-interactive contexts that is especially risky because the draft-Worker confirmation path uses the fallback yes behavior.

## Why this happens

The Worker-not-found fallback was originally built for create/update bulk uploads. After `null` delete support was added, delete-only input started sharing the same fallback path.

## Suggested fix

Only retry through `createDraftWorker()` when the bulk payload contains at least one non-null secret.

For delete-only input, Worker-not-found should remain an error.

## Repro

```sh
echo '{"OLD_SECRET": null}' | wrangler secret bulk --name typo-worker
```

### PR DESCRIPTION
**Pull request by matingathani**
Fixes #14052.

`wrangler secret bulk` now supports `null` values to delete secrets. However, when the payload is **delete-only** (all values are `null`) and the named Worker doesn't exist, the command incorrectly falls through to the Worker-not-found recovery path — which prompts to create a draft Worker. In non-interactive/CI contexts this auto-accepts, silently creating a Worker under a mistyped name.

This fix guards the `createDraftWorker` fallback: if every value in the bulk payload is `null`, the Worker-not-found error propagates unchanged instead of triggering Worker creation.

**Before:** `echo '{"OLD_SECRET": null}' | wrangler secret bulk --name typo-worker` → creates `typo-worker`  
**After:** same command → fails with the original Worker-not-found error

---

- Tests
  - [x] Tests included/updated
- Public documentation
  - [x] Documentation not necessary because: bug fix for edge case introduced by the recently-merged delete support; no user-visible API change

---

<a href="https://app.devin.ai/review/cloudflare/workers-sdk/pull/14058" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

**Comment by workers-devprod** [comment]
Codeowners approval required for this PR:
- @cloudflare/wrangler

<details><summary>Show detailed file reviewers</summary>

- .changeset/secret-bulk-delete-only-guard.md: [@cloudflare/wrangler]
- packages/wrangler/src/__tests__/secret.test.ts: [@cloudflare/wrangler]
- packages/wrangler/src/secret/index.ts: [@cloudflare/wrangler]

</details>

File: packages/wrangler/src/__tests__/secret.test.ts [hunk 1]
```diff
@@ -1594,6 +1594,32 @@ describe("wrangler secret", () => {
...
+				"secret.json",
+				JSON.stringify({
+					"secret-to-delete": null,
+				})
+			);
+			mockNoWorkerFound({ isBulk: true });
+
+			await expect(
+				runWrangler("secret bulk ./secret.json --name non-existent-worker")
+			).rejects.toThrow();
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/secret.test.ts]: I'd expect this test to assert on the "Worker not found" error

**Review by penalosa — COMMENTED** [review]
This is looking good, but the test could do with being a bit more specific. Thanks for the contribution!

**Comment by workers-devprod** [comment]
Codeowners approval required for this PR:
- @cloudflare/wrangler

<details><summary>Show detailed file reviewers</summary>

- .changeset/secret-bulk-delete-only-guard.md: [@cloudflare/wrangler]
- packages/wrangler/src/__tests__/secret.test.ts: [@cloudflare/wrangler]
- packages/wrangler/src/secret/index.ts: [@cloudflare/wrangler]

</details>

File: packages/wrangler/src/__tests__/secret.test.ts (hunk 1 above)
Comment (matingathani) [reply packages/wrangler/src/__tests__/secret.test.ts]: Fixed in 11aa84b — test now asserts `[APIError: A request to the Cloudflare API (...) failed.]` specifically, confirming the Worker-not-found error propagates correctly for delete-only payloads.

**Comment by matingathani** [comment]
Closing as duplicate of #14053 by @fallintoplace which was opened a day earlier. Apologies for not checking first.

**Comment by fallintoplace** [comment]
@matingathani All good Matin, my PR wasn't reviewed though. Feel free to take credits.
